finagle-mysql/src/main/scala/com/twitter/finagle/mysql/CanBeParameter.scala

```diff
@@ -0,0 +1,182 @@
+    }
+  }
+
+  implicit val booleanCanBeParameter = {
+    new CanBeParameter[Boolean] {
+      def sizeOf(param: Boolean) = 1
+      def typeCode(param: Boolean) = Type.Tiny
+      def write(writer: BufferWriter, param: Boolean) = writer.writeBoolean(param)
+    }
+  }
+
+  implicit val byteCanBeParameter = {
+    new CanBeParameter[Byte] {
+      def sizeOf(param: Byte) = 1
+      def typeCode(param: Byte) = Type.Tiny
+      def write(writer: BufferWriter, param: Byte) = writer.writeByte(param)
+    }
+  }
+
+  implicit val shortCanBeParameter = {
+    new CanBeParameter[Short] {
+      def sizeOf(param: Short) = 2
+      def typeCode(param: Short) = Type.Short
+      def write(writer: BufferWriter, param: Short) = writer.writeShort(param)
+    }
+  }
+
+  implicit val intCanBeParameter = {
+    new CanBeParameter[Int] {
+      def sizeOf(param: Int) = 4
+      def typeCode(param: Int) = Type.Long
+      def write(writer: BufferWriter, param: Int) = writer.writeInt(param)
+    }
+  }
+
+  implicit val longCanBeParameter = {
+    new CanBeParameter[Long] {
+      def sizeOf(param: Long) = 8
+      def typeCode(param: Long) = Type.LongLong
+      def write(writer: BufferWriter, param: Long) = writer.writeLong(param)
+    }
+  }
+
+  implicit val floatCanBeParameter = {
+    new CanBeParameter[Float] {
+      def sizeOf(param: Float) = 4
+      def typeCode(param: Float) = Type.Float
+      def write(writer: BufferWriter, param: Float) = writer.writeFloat(param)
+    }
+  }
+
+  implicit val doubleCanBeParameter = {
+    new CanBeParameter[Double] {
+      def sizeOf(param: Double) = 8
+      def typeCode(param: Double) = Type.Double
+      def write(writer: BufferWriter, param: Double) = writer.writeDouble(param)
+    }
+  }
+
+  implicit val byteArrayCanBeParameter = {
+    new CanBeParameter[Array[Byte]] {
+      def sizeOf(param: Array[Byte]) = Buffer.sizeOfLen(param.length) + param.length
+      def typeCode(param: Array[Byte]) = {
+        if (param.length <= 255) Type.TinyBlob
+        else if (param.length <= 65535) Type.Blob
+        else if (param.length <= 16777215) Type.MediumBlob
+        else -1
+      }
+      def write(writer: BufferWriter, param: Array[Byte]) = writer.writeLengthCodedBytes(param)
+    }
+  }
+
+  implicit val valueCanBeParameter = {
+    new CanBeParameter[Value] {
+      def sizeOf(param: Value) = param match {
+        case RawValue(_, _, true, b) => Buffer.sizeOfLen(b.length) + b.length
+        case StringValue(s)          => val bytes = s.getBytes(Charset.defaultCharset); Buffer.sizeOfLen(bytes.size) + bytes.size
+        case ByteValue(_)            => 1
+        case ShortValue(_)           => 2
+        case IntValue(_)             => 4
+        case LongValue(_)            => 8
+        case FloatValue(_)           => 4
+        case DoubleValue(_)          => 8
+        case NullValue               => 0
+        case _                       => 0
```

Comment:
@mosesn `Value` is a sealed trait and `typeCode` covers all of the cases explicitly. There is no case that's left out. In case of `sizeOf` I am just keeping the existing behavior as-is.